src/java/org/apache/cassandra/utils/JVMStabilityInspector.java

```diff
@@ -125,7 +123,31 @@ public static void inspectThrowable(Throwable t, boolean propagateOutOfMemory, C
             killer.killCurrentJVM(t);
 
         if (t.getCause() != null)
-            inspectThrowable(t.getCause(), propagateOutOfMemory, fn);
+            inspectThrowable(t.getCause(), fn);
+    }
+
+    /**
+     * Intentionally produce a heap space OOM upon seeing a Direct buffer memory OOM.
+     * Direct buffer OOM cannot trigger JVM OOM error related options,
+     * e.g. OnOutOfMemoryError, HeapDumpOnOutOfMemoryError, etc.
+     * See CASSANDRA-15214 for more details
+     */
+    @Exclude // Exclude from just in time compilation.
+    private static void forceHeapSpaceOomMaybe(OutOfMemoryError oom)
+    {
+        // See the oom thrown from java.nio.Bits.reserveMemory.
+        if (!"Direct buffer memory".equals(oom.getMessage()))
+        {
+            return;
+        }
+        logger.error("Force heap space OutOfMemoryError in the presence of", oom);
+        List<long[]> ignored = new ArrayList<>();
```

Comment:
It took me a sec to realize the goal here is to literally OOM the process. It might be nice to add a comment before the loop to that effect.